libdnf/module/ModuleSolver.cpp

```diff
@@ -0,0 +1,79 @@
+#include "ModuleSolver.hpp"
+
+#include <vector>
+#include <solv/solver.h>
+#include <iostream>
+#include <sstream>
+
+ModuleSolver::ModuleSolver(const std::shared_ptr<Pool> &pool)
+        : pool(pool)
+{
+    pool_createwhatprovides(pool.get());
+}
+
+void ModuleSolver::addModulePackage(const std::shared_ptr<ModulePackage> &modulePackage)
+{
+    modules.push_back(modulePackage);
+}
+
+std::vector<Id> ModuleSolver::solve()
+{
+    std::vector<Id> solvedIds;
+    pool_set_flag(pool.get(), POOL_FLAG_IMPLICITOBSOLETEUSESCOLORS, 1);
+
+    for (const auto &module : modules) {
+        Queue queue;
+        queue_init (&queue);
+
+        std::ostringstream ss;
+        ss << "module-ns(" << module->getName() << ":" << module->getStream() << ")";
```

Comment:
no, rpms belong to `module(n:s:v:c:a)` and not just `module(n:s)`

Reply:
It resolves only modules, not rpms.

Reply:
then why do you even do the selection? just do job install provides module(n:s)

Reply:
Doesn't the 'job install provides module(n:s)' return only the best match? We need to return all matching module versions.

Reply:
do you really need to check old versions? even those are unsupported and no, it doesn't return best match, it returns the transaction